fem/pnonlinearform.cpp

```diff
@@ -218,7 +218,8 @@ void ParBlockNonlinearForm::SetEssentialBC(const
 
 double ParBlockNonlinearForm::GetEnergy(const Vector &x) const
 {
-   xs_true.Update(x.GetData(), block_trueOffsets);
+   // xs_true is not modified, so const_cast is okay
+   xs_true.Update(const_cast<Vector &>(x), block_trueOffsets);
```

Comment:
Has the `Update` signature changed? Is there a reason for not using `GetData`?

Reply:
No changes in the signature of `BlockVector::Update` -- this is just switching to using the version of `BlockVector::Update` that uses a `Vector` to specify the "base". This version allows us to work correctly on host and device.